iceoryx_posh/source/runtime/posh_runtime_impl.cpp

```diff
@@ -372,10 +381,33 @@ NodeData* PoshRuntimeImpl::createNode(const NodeProperty& nodeProperty) noexcept
 }
 
 cxx::expected<InstanceContainer, FindServiceError>
-PoshRuntimeImpl::findService(const capro::IdString_t& service, const capro::IdString_t& instance) noexcept
+PoshRuntimeImpl::findService(const cxx::variant<SearchMode, capro::IdString_t> service,
+                             const cxx::variant<SearchMode, capro::IdString_t> instance) noexcept
 {
+    /// @todo #415 remove the string mapping, once the find call is done via shared memory
+    capro::IdString_t serviceString;
+    capro::IdString_t instanceString;
+
+    if (service.index() == 0U)
+    {
+        serviceString = "*";
+    }
+    else
+    {
+        serviceString = *service.get_at_index<1U>();
+    }
+
+    if (instance.index() == 0U)
+    {
+        instanceString = "*";
+    }
+    else
+    {
+        instanceString = *instance.get_at_index<1U>();
+    }
+
     IpcMessage sendBuffer;
-    sendBuffer << IpcMessageTypeToString(IpcMessageType::FIND_SERVICE) << m_appName << service << instance;
+    sendBuffer << IpcMessageTypeToString(IpcMessageType::FIND_SERVICE) << m_appName << serviceString << instanceString;
```

Comment:
That will be a limitation on the current `master`. However, this will be resolved in the next few weeks, so that things will behave just fine for the 2.0 :banana: release.

Reply:
That's okay, but can you add a todo for this, so that it will no be forgotten

Reply:
Have a look at line 387 ;)